### DISCUSSION TITLE
When clicking sign in button, How can I check if the username and password is valid or not without redirecting the sign in page from modal?

### DISCUSSION THREAD

**Discussion by FullStackDev365** [nextauthjs/next-auth#944] [Help]
When clicking sign in button, How can I check if the username and password is valid or not without redirecting the sign in page from modal?

Thanks for your efforts.
After searching the auth features related to next.js, I think this is the best and complete auth library.
Current I had a problem working with this next-auth library in my frontend written in next.js.
I know this next-auth library makes the sign in page automatically from the providers in [...nextauth].js file.
**I would like to sign in by calling api when clicking sign in button, without redirecting the sign in page.**
In other words, now I am going to show the sign in page as modal or by poping up it.
So when clicking the sign in button, I am going to check if the credentials is valid or not without redirecting the new sign in page from the modal.
I have researched all of the next-auth.org pages, but there are no things related to this problem.
Any helps will be very thankful.

**Comment by balazsorban44** [comment]
Would this PR solve your issue?: https://github.com/nextauthjs/next-auth/pull/1219
